pkg/pod/status.go

```diff
@@ -170,7 +170,15 @@ func setTaskRunStatusBasedOnStepStatus(logger *zap.SugaredLogger, stepStatuses [
 					logger.Errorf("error extracting the exit code of step %q in taskrun %q: %v", s.Name, tr.Name, err)
 					merr = multierror.Append(merr, err)
 				}
-				taskResults, pipelineResourceResults, filteredResults := filterResultsAndResources(results)
+				// collect results from taskspec
+				specResults := []v1beta1.TaskResult{}
+				if tr.Spec.TaskSpec != nil {
+					specResults = append(specResults, tr.Spec.TaskSpec.Results...)
+				}
+				if ts != nil {
+					specResults = append(specResults, ts.Results...)
```

Comment:
same point as above here.

Reply:
just thinking if you also need `&& ts.Results != nil` in the if condition.

Reply:
ts.Results is also a list, so it should be fine if we have an empty list

Reply:
SGTM!